### ISSUE TITLE
[injection] 2.3.8 broke certain regex match patterns

### ISSUE THREAD

**Issue by thehesiod** [aio-libs/aiohttp#2670]
## Long story short

https://github.com/aio-libs/aiohttp/pull/2667 causes the following scenario to break:

```python
router.add_get('/v1/grid/{variable}/{interval_type}/{date}', get_grid)
```
which compiles to:

`re.compile('\\/v1\\/grid\\/(?P<variable>[^{}/]+)\\/(?P<interval_type>[^{}/]+)\\/(?P<date>[^{}/]+)')`

which causes the following match group:
```python
<class 'tuple'>: ('temp', 'hourly', '2015-01-01T00:00:00%2B00:00')
```

and with this code that was recently modified in aiohttp:
```
{key: URL(value, encoded=True).path for key, value in
                    match.groupdict().items()}
```

yields

```python
{'variable': 'temp', 'interval_type': 'hourly', 'date': '00:00+00:00'}
```

## Expected behaviour

should not strip out the date from the datetime string

## Actual behaviour

date is getting stripped out

## Your environment

aiohttp 2.3.8
yarl 1.0.0

**Comment by asvetlov** [comment]
Reproduced.
Will make bugfix release today

**Comment by asvetlov** [comment]
Fixed by 2.3.9 release
